### DISCUSSION TITLE
CMake build with add_subdirectory

### DISCUSSION THREAD

**Discussion by Please-just-dont** [jrouwe/JoltPhysics#1305] [Q&A]
I noticed there's about 10 different build instructions depending on what platform you're on and what compiler you're using. Usually with a CMake project you can simply add_subdirectory and link to the target, and it (hopefully) should work everywhere. Same pipeline for mobile, desktop etc. Is there a possibility of this becoming a thing in the future?

**Comment by jrouwe** [comment]
The 10 different build instructions all boil down to the same thing: run cmake to build. The way you instruct cmake to build varies slightly from platform to platform because on one you use makefiles, on another you want visual studio project files etc. I haven't tried `add_subdirectory` but cmake fetchcontent should [work](https://github.com/jrouwe/JoltPhysicsHelloWorld/blob/5434026068623e83abff9c6d6894582349807ea2/Build/CMakeLists.txt#L55-L61) on all platforms. I would expect `add_subdirectory` to work too (maybe with some minor tweaks, if that's needed let me know).
